fineract-provider/src/main/java/org/apache/fineract/portfolio/loanproduct/serialization/LoanProductDataValidator.java

```diff
@@ -345,6 +346,22 @@ public void validateForCreate(final JsonCommand command) {
         // Validating whether the processor is existing
         loanRepaymentScheduleTransactionProcessorFactory.determineProcessor(transactionProcessingStrategyCode);
 
+        String loanScheduleProcessingType = LoanScheduleProcessingType.HORIZONTAL.name();
```

Comment:
Why did you move these checks?

Reply:
In order to have the input value at that point and reuse the current validation method